src/data/finset/basic.lean

```diff
@@ -1945,6 +1955,20 @@ lemma range_add_one' (n : ℕ) :
   range (n + 1) = insert 0 ((range n).map ⟨λi, i + 1, assume i j, nat.succ.inj⟩) :=
 by ext (⟨⟩ | ⟨n⟩); simp [nat.succ_eq_add_one, nat.zero_lt_succ n]
 
+lemma range_add_eq_union (m n : ℕ) :
+  range (m + n) = range m ∪ (range n).map (add_left_embedding m) :=
```

Comment:
This should follow from some generic lemmas about `finset.Ico` (and `finset.Ico 0 m = finset.range m`).